ckanext/datagovtheme/fanstatic_library/scripts/geoplatform.js

```diff
@@ -0,0 +1,26 @@
+async function getGeoplatformLink() {
+    let parentEl = document.getElementById("geoplatform-link-section");
+    if (parentEl == null) {
+        return;
+    }
+    let datasetName = parentEl.getAttribute("data-package-name");
+    try {
+        const response = await fetch(
+            `https://stg-api.geoplatform.gov/v3/public/lookups/data-gov/dataset?name=${datasetName}`
+        )
+            .then((response) => {
+                if (response.ok) {
+                    return response.json();
+                }
+            })
+            .then((data) => {
+                let el = document.getElementById("geoplatform-link");
+                el.href = data.geoplatform_url;
+                parentEl.classList.remove("hide");
+            });
+    } catch (error) {
+        console.error(error);
+    }
+}
+
+getGeoplatformLink();
```

Comment:
should this getGeoplatformLink() be called only after all dom elements is loaded and ready?

Reply:
since we do SSR of all templates, is there a need to consider this?

Reply:
Looking at the final js code, the function is called when it is loaded, regardless the DOM state. If we are sure the js file is always loaded at the bottom the files, after the DOM element in question, we might not have to add any code. But to be safe, we can simply wrap everthing inside a 
```js
jQuery(function($){
  //all goes inside.
});
```

Reply:
File should be loaded at the bottom of the page, but if we want to be safe, we can wrap it in a $docReady like the above. 

There's also this which is pure JS, but since we have jQuery already why not make use of it.

```
document.addEventListener('DOMContentLoaded', function(){ 
// your code goes here
}, false);
```

https://gist.github.com/flashvnn/18901998fc9b42898bab69ec1abadf85

Reply:
include [here](https://github.com/GSA/ckanext-datagovtheme/pull/178/commits/3cb778ffa5d89fa85aef5e777504d8dbdd3cdfbf)